dataRetriever/requestHandlers/requestHandler.go

```diff
@@ -358,18 +363,29 @@ func (rrh *resolverRequestHandler) RequestTrieNodes(destShardID uint32, hashes [
 	rrh.mutexTrieHashes.Lock()
 	defer rrh.mutexTrieHashes.Unlock()
 
-	rrh.trieHashAccumulator = append(rrh.trieHashAccumulator, unrequestedHashes...)
+	for _, hash := range unrequestedHashes {
+		rrh.trieHashesAccumulator[string(hash)] = struct{}{}
+	}
+
+	index := 0
+	itemsToRequest := make([][]byte, len(rrh.trieHashesAccumulator))
+	for hash := range rrh.trieHashesAccumulator {
+		itemsToRequest[index] = []byte(hash)
+		index++
+	}
+
+	rrh.whiteList.Add(itemsToRequest)
+	rrh.addRequestedItems(itemsToRequest)
```

Comment:
Move this in line 416 to be done only if the request of these items is done

Reply:
moved addRequestedItems - whitelisting would leave to the top.